Sources/BrowserServicesKit/ContentBlocking/TrackerDataQueryExtension.swift

```diff
@@ -34,6 +34,23 @@ extension TrackerData {
         return nil
     }
 
+    /// Returns the parent of the entity associated with the host if any. If the entity associated with the host doesn't have a parent return the entity. Otherwise, return nil.
+    /// - Parameter host:
+    /// - Returns: The entity associated with the host.
+    public func findParentEntityOrFallback(forHost host: String) -> Entity? {
```

Comment:
Perhaps I can just modify instead of creating a new one. 

```
public func findEntity(forHost host: String) -> Entity? {
        for host in variations(of: host) {
            if let entityName = domains[host] {
                return entities[entityName]
            }
        }
        return nil
    }

```

`TrackerResolver` & `ReferrerTrimming` use `findEntity(forHost` and I’m not sure if we always want to find the parent entity if it exists. I think this is safer?